motionplan/plan.go

```diff
@@ -307,3 +307,37 @@ func (e *ExecutionState) CurrentInputs() map[string][]referenceframe.Input {
 func (e *ExecutionState) CurrentPoses() map[string]*referenceframe.PoseInFrame {
 	return e.currentPose
 }
+
+// CalculateFrameErrorState takes an ExecutionState and a Frame and calculates the error between the Frame's expected
+// and actual positions.
+func CalculateFrameErrorState(e ExecutionState, frame referenceframe.Frame) (spatialmath.Pose, error) {
+	currentInputs, ok := e.CurrentInputs()[frame.Name()]
+	if !ok {
+		return nil, fmt.Errorf("could not find frame %s in ExecutionState", frame.Name())
+	}
+	currentPose, ok := e.CurrentPoses()[frame.Name()]
+	if !ok {
+		return nil, fmt.Errorf("could not find frame %s in ExecutionState", frame.Name())
+	}
+	currPoseInArc, err := frame.Transform(currentInputs)
+	if err != nil {
+		return nil, err
+	}
+	path := e.Plan().Path()
+	if path == nil {
+		return nil, errors.New("cannot calculate error state on a nil Path")
```

Comment:
If there is no plan, currently we will return a zero pose in Error State. We probably want to continue this.